Swiften/Elements/MIXJoin.h

```diff
@@ -43,16 +43,20 @@ namespace Swift {
                 jid_ = jid;
             }
 
-            const std::vector<MIXSubscribe::ref>& getSubscriptions() const {
+            const std::unordered_set<std::string>& getSubscriptions() const {
                 return subscribeItems_;
             }
 
-            void setSubscriptions(const std::vector<MIXSubscribe::ref>& value) {
+            void setSubscriptions(const std::unordered_set<std::string>& value) {
                 subscribeItems_ = value ;
             }
 
-            void addSubscription(MIXSubscribe::ref value) {
-                subscribeItems_.push_back(value);
+            void addSubscription(std::string value) {
+                subscribeItems_.insert(value);
+            }
+
+            bool hasSubscription(std::string value) {
```

Comment:
This should `const std::string& value`, as you don't copy the parameter to a member or local variable. Also, this method can be const.